server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourceContext.java

```diff
@@ -34,12 +40,25 @@ public class FetchSourceContext implements Writeable, ToXContentObject {
 
     public static final ParseField INCLUDES_FIELD = new ParseField("includes", "include");
     public static final ParseField EXCLUDES_FIELD = new ParseField("excludes", "exclude");
+    public static final ParseField INCLUDE_VECTORS = new ParseField("include_vectors");
 
-    public static final FetchSourceContext FETCH_SOURCE = new FetchSourceContext(true, Strings.EMPTY_ARRAY, Strings.EMPTY_ARRAY);
-    public static final FetchSourceContext DO_NOT_FETCH_SOURCE = new FetchSourceContext(false, Strings.EMPTY_ARRAY, Strings.EMPTY_ARRAY);
+    public static final boolean DEFAULT_INCLUDE_VECTORS = Boolean.TRUE;
+    public static final FetchSourceContext FETCH_SOURCE = new FetchSourceContext(
+        true,
+        Strings.EMPTY_ARRAY,
+        Strings.EMPTY_ARRAY,
+        DEFAULT_INCLUDE_VECTORS
+    );
+    public static final FetchSourceContext DO_NOT_FETCH_SOURCE = new FetchSourceContext(
+        false,
+        Strings.EMPTY_ARRAY,
+        Strings.EMPTY_ARRAY,
+        DEFAULT_INCLUDE_VECTORS
+    );
     private final boolean fetchSource;
     private final String[] includes;
     private final String[] excludes;
+    private final boolean includeVectors;
```

Comment:
I thought we needed to use a `Boolean` here so that we can use `null` here as well? My understanding was:

`true`: Include all vectors (`semantic_text`, `dense_vector`, `sparse_vector`)
`null`: Exclude vectors only for `semantic_text`
`false`: Exclude all vectors (`semantic_text`, `dense_vector`, `sparse_vector`)

@carlosdelest Is that correct?

Reply:
Correct!

Reply:
Ideally yes but if we set `null` as the default value, `reindex` and others(probably `update_api`, `mget`) will also start to exclude embedding from `semantic_text` fields. It looked like we wanted to have a default value of `null` for `search` operations and a default value of `true` for `reindex` and other APIs which is quite difficult to maintain. That is why I switched to handling only `true` and `false` values.

Are we open to the idea of modifying(force to set the value to `true`) those APIs (reindex, mget, update)? I am open to other suggestions too.

cc: @Mikep86 @carlosdelest @kderusso

Reply:
I think we can maintain different values for different code paths fairly easily. I haven't looked at each code path in detail yet, but we can use a mix of these approaches to achieve what we want:

- If the code path involves parsing a user-provided `_source`, we can add an optional `defaultIncludeVectors` param to `fromXContent`.
- If the code path involves programmatically building a `FetchSourceContext`, we can use the `of` method that allows us to specify the `includeVectors` value we want.

The Update API requires a bit more investigation because at first glance it's unclear what `_source` is used for. The update response does not return any document source, so it must be used in the actual update operation. We need to dig a bit deeper on that one to determine if we want to disallow filtering vectors on that code path. My intuition tells me that we do, but we need to confirm and then we can decide on the course of action for that particular path.

Reply:
I don't think we ever want to exclude vectors outside of a search.

Reply:
> I don't think we ever want to exclude vectors outside of a search.

Disagree here. I think we want to exclude vectors by default in most (if not all) cases, except where it is required to perform the desired operation (i.e. reindex and update_by_query so far, where the document source isn't even returned to the user). This creates a uniform experience for the user and prevents responses to requests like document get & multi-get being polluted with semantic text embeddings by default.

Reply:
Good point, I agree with getting a document. But I think that searching and document _retrieval_ are the only use cases we should support this for. We should add additional cases as we see fit, but the default internal behavior and for calls that update state should always be to include vectors. That will include updates and reindexes for example.

Reply:
Agreed, that makes sense. So far, the cases that I am aware of that use source to update state are:

- Reindex
- Update by query
- Update (maybe? Requires a bit more investigation)

Any others that I'm missing? The Bulk API also accepts `_source`, but according to the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html#docs-bulk-api-query-params) its used to control the fields that are returned.

Reply:
Without doing a deep dive those are the only others I am aware of.

Reply:
We can remove the Update APi from that list, see my comment here: https://github.com/elastic/elasticsearch/pull/112319#discussion_r1744337968